go/runtime/bundle/registry.go

```diff
@@ -152,6 +152,37 @@ func (r *Registry) AddManifest(manifest *Manifest, dir string) error {
 	return nil
 }
 
+// Manifests returns all known manifests (regular and detached).
+func (r *Registry) Manifests() []ExplodedManifest {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	return slices.Collect(maps.Values(r.manifests))
+}
+
+// RemoveManifest removes a manifest with provided hash.
+func (r *Registry) RemoveManifest(hash hash.Hash) bool {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	explManifest, ok := r.manifests[hash]
+	if !ok {
+		return false
+	}
+
+	runtimeID := explManifest.Manifest.ID
+	r.logger.Info("Removing manifest from registry",
+		"id", runtimeID,
+		"manifest_hash", hash,
+	)
+	if ronl := explManifest.GetComponentByID(component.ID_RONL); ronl != nil {
+		delete(r.regularManifests[runtimeID], ronl.Version)
+	}
+	delete(r.manifests, hash)
```

Comment:
I would put this 3 lines higher, to be inline with the order of struct fields, plus it makes code more readable.